crates/columns/src/lib.rs

```diff
@@ -0,0 +1,567 @@
+            return quote! {
+                fields.push(format!("{}", #index));
+            };
+        }
+    }
+
+    // Check for Vec<T>
+    if is_vec_of_generic(field_type, generic_type) {
+        return quote! {
+            // Add single index representation for vectors
+            fields.push(format!("{}__idx", #index));
+        };
+    }
+
+    // Handle `GenericArray<ElementType, Size>`
+    if let syn::Type::Path(type_path) = field_type {
+        if type_path.path.segments.last().unwrap().ident == "GenericArray" {
+            if let syn::PathArguments::AngleBracketed(args) =
+                &type_path.path.segments.last().unwrap().arguments
+            {
+                let element_type = &args.args[0];
+                let size_type = &args.args[1];
+                let element_type = match args
+                    .args
+                    .first()
+                    .expect("Expected element type in GenericArray")
+                {
+                    syn::GenericArgument::Type(ty) => ty,
+                    _ => panic!("Expected a type as the first argument in GenericArray"),
+                };
+
+                let size_type = match args
+                    .args
+                    .iter()
+                    .nth(1)
+                    .expect("Expected size type in GenericArray")
+                {
+                    syn::GenericArgument::Type(ty) => ty,
+                    _ => panic!("Expected a type as the second argument in GenericArray"),
+                };
+
+                // Check if the element type is `T`
+                if let syn::Type::Path(type_path) = element_type {
+                    if type_path.path.is_ident(&generic_type.ident) {
+                        // Treat the `GenericArray` of `T` as terminal
+                        return quote! {
+                            for i in 0..<#size_type as typenum::Unsigned>::USIZE {
+                                fields.push(format!("{}__{}", #index, i));
+                            }
+                        };
+                    }
+                }
+
+                return quote! {
+                    if let Some(sub_fields) = <#element_type as FlattenFieldsHelper>::flatten_fields() {
+                        for i in 0..<#size_type as typenum::Unsigned>::USIZE {
+                            for sub_field in &sub_fields {
+                                fields.push(format!("{}__{}__{}", #index, i, sub_field));
+                            }
+                        }
+                    } else {
+                        for i in 0..<#size_type as typenum::Unsigned>::USIZE {
+                            fields.push(format!("{}__{}", #index, i));
+                        }
+                    }
+                };
+            }
+        }
+    }
+
+    // Special handling for MemoryAddress<T, T> where T is the generic parameter
+    if let syn::Type::Path(type_path) = field_type {
```

Comment:
I added specific cases for `MemoryAddress` and `ExecutionState`, where I have to hardcode the columns. Please check column names :)